WalletWasabi/Nito/AsyncEx/AsyncMutex.cs

```diff
@@ -401,18 +401,29 @@ public static async Task WaitForAllMutexToCloseAsync()
 				}
 			}
 
+			var lastLog = DateTime.MinValue;
+			var lastNumberOfAliveMutexes = 0;
+
 			while (true)
 			{
 				lock (AsyncMutexesLock)
 				{
-					bool stillRunning = AsyncMutexes.Any(am => am.Value.IsAlive);
-					if (!stillRunning)
+					int numberOfAliveMutexes = AsyncMutexes.Count(am => am.Value.IsAlive);
+					if (numberOfAliveMutexes == 0)
 					{
 						return;
 					}
-					Logger.LogDebug($"Waiting for: {string.Join(", ", AsyncMutexes.Where(am => am.Value.IsAlive).Select(m => m.Value.ShortName))}.");
+
+					// Log every 10 second or status change.
+					if ((DateTime.Now - lastLog) > TimeSpan.FromSeconds(10) || lastNumberOfAliveMutexes != numberOfAliveMutexes)
```

Comment:
Do not use `Now`. `Now` should be only use to display the date and it is incredible slow. Use `UtcNow` instead.

Reply:
+1